exchanges/bybit/bybit.go

```diff
@@ -8,740 +8,2348 @@ import (
+		params.Set("settleCoin", settleCoin)
+	}
+	if orderID != "" {
+		params.Set("orderId", orderID)
+	}
+	if orderLinkID != "" {
+		params.Set("orderLinkId", orderLinkID)
+	}
+	if openOnly != 0 {
+		params.Set("openOnly", strconv.FormatInt(openOnly, 10))
+	}
+	if orderFilter != "" {
+		params.Set("orderFilter", orderFilter)
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	if cursor != "" {
+		params.Set("cursor", cursor)
+	}
+	var resp *TradeOrders
+	return resp, by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodGet, "/v5/order/realtime", params, nil, &resp, getOrderEPL)
+}
+
+// CancelAllTradeOrders cancel all open orders
+func (by *Bybit) CancelAllTradeOrders(ctx context.Context, arg *CancelAllOrdersParam) ([]OrderResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	err := isValidCategory(arg.Category)
+	if err != nil {
+		return nil, err
+	}
+	if by.AccountType == accountTypeNormal &&
+		arg.Category == cOption {
+		return nil, errAPIKeyIsNotUnified
+	}
+	var resp CancelAllResponse
+	return resp.List, by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodPost, "/v5/order/cancel-all", nil, arg, &resp, calcelAllEPL)
+}
+
+// GetTradeOrderHistory retrieves order history. As order creation/cancellation is asynchronous, the data returned from this endpoint may delay.
+// If you want to get real-time order information, you could query this endpoint or rely on the websocket stream (recommended).
+func (by *Bybit) GetTradeOrderHistory(ctx context.Context, category, symbol, orderID, orderLinkID,
+	baseCoin, settleCoin, orderFilter, orderStatus, cursor string,
+	startTime, endTime time.Time, limit int64) (*TradeOrders, error) {
+	params, err := fillCategoryAndSymbol(category, symbol, true)
+	if err != nil {
+		return nil, err
+	}
+	if by.AccountType == accountTypeNormal &&
+		category == cOption {
+		return nil, errAPIKeyIsNotUnified
+	}
+	if baseCoin != "" {
+		params.Set("baseCoin", baseCoin)
+	}
+	if settleCoin != "" {
+		params.Set("settleCoin", settleCoin)
+	}
+	if orderID != "" {
+		params.Set("orderId", orderID)
+	}
+	if orderLinkID != "" {
+		params.Set("orderLinkId", orderLinkID)
+	}
+	if orderFilter != "" {
+		params.Set("orderFilter", orderFilter)
+	}
+	if orderStatus != "" {
+		params.Set("orderFilter", orderFilter)
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	if cursor != "" {
+		params.Set("cursor", cursor)
+	}
+	if !startTime.IsZero() {
+		params.Set("startTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+	}
+	if !endTime.IsZero() {
+		params.Set("endTime", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
+	var resp *TradeOrders
+	return resp, by.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodGet, "/v5/order/history", params, nil, &resp, getOrderHistoryEPL)
+}
+
+// PlaceBatchOrder place batch or trade order.
+func (by *Bybit) PlaceBatchOrder(ctx context.Context, arg *PlaceBatchOrderParam) ([]BatchOrderResponse, error) {
```

Comment:
`BatchOrderParam` is missing request data
https://bybit-exchange.github.io/docs/v5/order/batch-place